besu/src/main/java/org/hyperledger/besu/controller/BesuController.java

```diff
@@ -361,7 +359,7 @@ public BesuControllerBuilder fromGenesisFile(
       // wrap with TransitionBesuControllerBuilder if we have a terminal total difficulty:
       if (configOptions.getTerminalTotalDifficulty().isPresent()) {
         // Enable start with vanilla MergeBesuControllerBuilder for PoS checkpoint block
-        if (isCheckpointSync(syncMode) && isCheckpointPoSBlock(configOptions)) {
+        if (syncMode == SyncMode.CHECKPOINT && isCheckpointPoSBlock(configOptions)) {
```

Comment:
```suggestion
        if (SyncMode.CHECKPOINT == syncMode  && isCheckpointPoSBlock(configOptions)) {
```

nit: this way around avoids a null pointer if syncMode were ever to be null (which shouldn't happen to be honest)

Reply:
I thought if syncMode were null, it should just evaluate `null == SyncMode.CHECKPOINT`, which should be ok and return false as expected, right?

Reply:
Sorry, I was getting confused with `nullableField.equals(value)` idiom throwing a null pointer. 
I should have suggested SyncMode.CHECKPOINT.equals(syncMode). 

You're right, both will return false so it's not a big deal, mostly stylistic.